### PR TITLE
*: Allow listening locally and specifying additional containers

### PR DESCRIPTION
**Pull request by brancz**
This will allow one to specify a proxy that is actually the entry point for a Prometheus/Alertmanager Pods UI.

@ant31 @fabxc

### CODE REVIEW

**Comment by brancz** [comment]
@gianrubio any idea why the helm hashes are different?

**Comment by gianrubio** [comment]
Hum... it's because your branch it's not rebased to master and the script it's not smart enough :) Do you mind to rebase it or to temporary remove the e2e-helm step?
As soon I have time I'll definitely fix that helm e2e script

**Comment by brancz** [comment]
@gianrubio hmm .. I rebased but it doesn't seem to have fixed the issue.

File: pkg/alertmanager/statefulset.go [hunk 1]
```diff
@@ -200,6 +205,24 @@ func makeStatefulSetSpec(a *monitoringv1.Alertmanager, config Config) (*v1beta1.
 		},
 	}
 
+	var livenessProbe *v1.Probe
+	var readinessProbe *v1.Probe
+	if !a.Spec.ListenLocal {
```
Comment (ant31) [inline pkg/alertmanager/statefulset.go]: why removing the probe?

Comment (brancz) [reply pkg/alertmanager/statefulset.go]: probes on loopback don't seem to work, so I decided it's the responsibility of the "entry point" container

**Comment by brancz** [comment]
Note there are some problems with tests on master, I am fixing those before I move forward with this PR. I'll comment here once the issues on master are resolved.

**Pull request merged**